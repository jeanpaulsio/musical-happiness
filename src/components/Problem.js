import React from "react";
import { func, string } from "prop-types";

Problem.propTypes = {
  title: string,
  render: func
}

function Problem(props) {
  return (
    <section>
      <h2>{props.title}</h2>
      {props.render()}
    </section>
  );
}

export default Problem;
