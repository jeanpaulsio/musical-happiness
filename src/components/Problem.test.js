import React from "react";
import renderer from "react-test-renderer";
import Problem from "./Problem";

test("<Problem /> renders correctly", () => {
  const title = "Problem 1";
  const tree = renderer
    .create(<Problem title={title} render={() => <h1 />} />)
    .toJSON();

  expect(tree.children.length).toEqual(2);
  expect(tree).toMatchSnapshot();
});
