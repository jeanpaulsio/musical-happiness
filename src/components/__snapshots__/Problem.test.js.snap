// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Problem /> renders correctly 1`] = `
<section>
  <h2>
    Problem 1
  </h2>
  <h1 />
</section>
`;
